addons/html_editor/static/src/components/history_dialog/history_dialog.scss

```diff
@@ -1,168 +1,261 @@
-                z-index: 10;
+                font-size: .5rem;
+                padding-left: 2px;
+                color: white;
+                background-color: $o-action;
             }
-            &.targeted {
-                color: lighten($primary, 20%);
-                &:before {
-                    color: white;
-                    content: '\f00c';
-                    background-color: lighten($primary, 20%);
-                }
-                &:after {
-                    border-color: lighten($primary, 20%);
-                }
+           
+            ~ .list-group-item:before {
+                background-color: $o-action;
             }
-            &.selected {
-                color: $primary;
-                &:before {
-                    color: white;
-                    content: '\f0da';
-                    background-color: $primary;
-                }
-                &:after {
-                    border-color: $primary;
-                }
-                .o_avatar {
-                    opacity: 1;
+        }
+
+        &:first-child:after {
+            content: none;
+        }
+    }
+}
+
+@include media-breakpoint-down(md) {
+    .html-history-dialog-container {
+        .nav-tabs {
+            background-color: $o-view-background-color;
+
+            .nav-link {
+                background-color: $o-gray-100;
+                border-bottom-color: $border-color;
+
+                &.active {
+                    --#{$prefix}nav-tabs-link-active-color: var(--historyDialog__link-color--active, #{$o-brand-odoo});
+                    border-bottom-color: $o-view-background-color;
+
+                    &::after {
+                        @include o-position-absolute($top: -$border-width, $right: -$border-width, $left: -$border-width);
+                        height: $border-width * 3;
+                        background-color: var(--historyDialog__link-border-top-color--active, #{$o-brand-odoo});
+                        content: '';
+                    }
                 }
             }
-            &:first-child:after {
-                content: none !important;
+        }
+    }
+
+    .html-history-dialog {
+        flex-direction: column;
+
+        .o_revision_panel {
+            min-width: 100%;
+            width: 100%;
+            flex: 1;
+            border-right: none;
+            overflow-y: auto;
+        }
+
+        .history-container {
+            flex: 1;
+            min-height: 0;
+        }
+
+        &.mobile-show-revisions {
+            .o_revision_panel {
+                display: block;
+            }
+            .history-container {
+                display: none !important;
             }
+        }
 
+        &.mobile-show-content {
+            .o_revision_panel {
+                display: none;
+            }
+            .history-container {
+                display: flex;
+                flex-direction: column;
+            }
         }
```

Comment:
Can we use `isMobileActiveTab` instead of `getMobileDialogClass()`, we could apply condionnal classes on `o_revision_panel` and `history-container` with the isMobileActiveTab props instead and get rid of the function and custom style for utilities: `d-none` `d-md-flex`.